drivers/dma/dma_iproc_pax.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef DMA_IPROC_PAX
+#define DMA_IPROC_PAX
+
+#include "dma_iproc_pax_common.h"
+
+/* Register RM_CONTROL fields */
+#define RM_CONTROL_MODE_MASK 0x3
+#define RM_CONTROL_MODE_SHIFT 0
+#define RM_CONTROL_MODE_TOGGLE 0x2
+#define RM_CONTROL_CONFIG_DONE BIT(2)
+#define RM_CONTROL_CONFIG_DONE_SHIFT 2
+#define RM_CONTROL_CONFIG_DONE BIT(4)
+#define RM_CONTROL_LINE_INTR_EN_SHIFT 4
+#define RM_CONTROL_LINE_INTR_EN BIT(4)
+#define RM_CONTROL_AE_TIMEOUT_EN_SHIFT 5
+#define RM_CONTROL_AE_TIMEOUT_EN BIT(5)
+
+#define TYPE_DMA_DESC 0x3
+#define NUM_BD_BUFFS 8
+/* DMA desc count: 3 entries per packet */
+#define RM_DESC_BDCOUNT 3
+/* 1 DMA packet desc takes 3 BDs */
+#define DMA_DESC_SIZE (RM_DESC_BDWIDTH * RM_DESC_BDCOUNT)
+/* Max size of transfer in single packet */
+#define MAX_DMA_SIZE_PER_BD (16 * 1024 * 1024)
+
+/* ascii signature  'V' 'K' */
+#define WRITE_SYNC_SIGNATURE 0x564B
+
+/* DMA transfers supported from 4 bytes thru 16M, size aligned to 4 bytes */
+#define PAX_DMA_MIN_SIZE 4
+#define PAX_DMA_MAX_SIZE (16 * 1024 * 1024)
+
+/* Bits 0:1 ignored by PAX DMA, i.e. 4-byte address alignment */
+#define PCI_ADDR_LS_IGNORE_BITS 2
+#define PCI_ADDR_ALIGNMT_SHIFT PCI_ADDR_LS_IGNORE_BITS
+/* PCI MS[1:0] shifted to LS[31:30] */
+#define PCI_ADDR_MS_TO_LS_SHIFT 30
+#define PCI_ADDR_MS_TO_LS_MASK 0x3
+
+#define PCI_ADDR_LO(pci) (((pci).addr_lo >> PCI_ADDR_ALIGNMT_SHIFT) | \
+			  (((pci).addr_hi) & PCI_ADDR_MS_TO_LS_MASK) << \
+			  PCI_ADDR_MS_TO_LS_SHIFT)
+
+#define PCI_ADDR_HI(pci) ((pci).addr_hi >> PCI_ADDR_ALIGNMT_SHIFT)
+
+/* RM header desc field */
+struct rm_header {
+	uint64_t opq : 16; /*pkt_id 15:0*/
+	uint64_t res1 : 20; /*reserved 35:16*/
+	uint64_t bdcount : 5; /*bdcount 40:36*/
+	uint64_t prot : 2; /*prot 41:40*/
+	uint64_t res2 : 13; /*reserved 55:43*/
+	uint64_t start : 1; /*start pkt :56*/
+	uint64_t end : 1; /*end pkt :57*/
+	uint64_t toggle : 1; /*toggle :58*/
+	uint64_t res3 : 1; /*reserved :59*/
+	uint64_t type : 4; /*type 63:60*/
+} __attribute__ ((__packed__));
```

Comment:
It looks like your naming is "..pax.h" for v1, "...pax_v2.h" for v2, and "...pax_common.h" for shared pieces -- it's slightly confusing compared to using "...pax_v1.h" and then "...pax.h" for shared. But that's just a minor comment.

The other concern is the duplication of definitions, with structs having different members when they come from different files, instead of being in one file where it is more obvious that there are two variants. But even then, it seems like the v2 variants are supersets, i.e. they carry forward the same fields but make use of some of the reserved bits. Would it make sense to use the v2 structs everywhere, just not touching the v2 fields in the v1 code paths?

Reply:
Yes, having "..pax_v1.h" and "..pax_v2.h" is better for clear understanding.

V1 and V2 RM hardware versions have different BD formats, so kept the different definitions.
V1 and V2 needs slightly different programming sequence as well, so the idea of having different driver files.